shark/shark_importer.py

```diff
@@ -0,0 +1,112 @@
+    self.device = device
+    self.dynamic = dynamic
+    self.jit_trace = jit_trace
+    self.benchmark_mode = benchmark_mode
+    self.inputs = None
+    self.input_details = None
+    self.output_details = None
+
+    # create tmp model file directory
+    if self.model_path is None:
+      print("Error. No model_path, Please input model path.")
+      return
+
+    if self.model_source_hub == "tfhub":
+      # compile and run tfhub tflite
+      if self.model_type == "tflite":
+        print("Setting up for TMP_DIR")
+        exe_basename = os.path.basename(sys.argv[0])
+        self.workdir = os.path.join(os.path.dirname(__file__), "tmp", exe_basename)
+        print(f"TMP_DIR = {self.workdir}")
+        os.makedirs(self.workdir, exist_ok=True)
+        self.tflite_file = '/'.join([self.workdir, 'model.tflite'])
+        print("Setting up local address for tflite model file: ", self.tflite_file)
+        if os.path.exists(self.model_path):
+          self.tflite_file = self.model_path
+        else:
+          print("Download tflite model")
+          urllib.request.urlretrieve(self.model_path, self.tflite_file)
+        print("Setting up tflite interpreter")
+        self.tflite_interpreter = tf.lite.Interpreter(model_path=self.tflite_file)
+        self.tflite_interpreter.allocate_tensors()
+        # default input initialization
+        self.input_details, self.output_details = self.get_model_details()
+        inputs = self.generate_inputs(self.input_details)  # device_inputs
+        self.setup_inputs(inputs)
+
+  def generate_inputs(self, input_details):
+    args = []
+    for input in input_details:
+      print(str(input["shape"]), input["dtype"].__name__)
+      args.append(np.zeros(shape=input["shape"], dtype=input["dtype"]))
+    return args
+
+  def get_model_details(self):
+    if self.model_type == "tflite":
+      print("Get tflite input output details")
+      self.input_details = self.tflite_interpreter.get_input_details()
+      self.output_details = self.tflite_interpreter.get_output_details()
+      return self.input_details, self.output_details
+
+  def setup_inputs(self, inputs):
+    print("Setting up inputs")
+    self.inputs = inputs
+
+  def compile(self):
+    # preprocess model_path to get model_type and Model Source Hub
+    print("Shark Importer compile Model")
+    if self.model_source_hub == "tfhub":
+      # compile and run tfhub tflite
+      print("Inference tfhub model")
+      self.shark_module = SharkInference(self.tflite_file, self.inputs,
+                                    device=self.device,
+                                    dynamic=self.dynamic,
+                                    jit_trace=self.jit_trace)
+      self.shark_module.set_frontend("tflite")
+      self.shark_module.compile()
+    elif self.model_source_hub == "huggingface":
+      print("Inference huggingface model")
+    elif self.model_source_hub == "jaxhub":
+      print("Inference JAX hub model")
```

Comment:
This is not doing anything yet right? Can we modify the print to, "Inference <model_source_hub_name> not implemented yet"?

Reply:
Done

Reply:
Looks great, thanks! :)